File: predict_pipeline.py
import timeit

import pandas as pd
from sklearn.datasets import fetch_openml

from train import train


def load_dataset():
    X, y = fetch_openml("titanic", version=1, as_frame=True, return_X_y=True, parser='liac-arff')
    return X


def do_predict(pipeline, X):
    return pipeline.predict_proba(X[['pclass', 'sex', 'age', 'fare']])[:, 0].astype(float).tolist()


def py_benchmark():
    pipeline = train()
    X, y = fetch_openml("titanic", version=1, as_frame=True, return_X_y=True, parser='liac-arff')

    def evaluate_predict():
        do_predict(pipeline, X)

    t = timeit.Timer(lambda: evaluate_predict())
    out = t.repeat(repeat=100, number=1)
    print((pd.DataFrame({'t, milliseconds': out}) * 1000).describe())

File: predict_pipeline.R
library(reticulate)
use_python("/usr/bin/python3")
source_python("train.py")

pipeline <- train()
source_python("predict_pipeline.py")

X <- load_dataset()

library("microbenchmark")
microbench_out <- microbenchmark(do_predict(pipeline, X))
microbench_out

py_benchmark()
